src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useEffect } from 'react';
+import Dialog from '@mui/material/Dialog';
+import { DialogProps, Theme } from '@mui/material';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import KeyboardArrowLeftIcon from '@mui/icons-material/KeyboardArrowLeft';
+import CloseIcon from '@mui/icons-material/Close';
+
+import Box from '../Box';
+import Button from '../Button';
+import IconButton from '../IconButton';
+import Typography from '../Typography';
+
+export type ConfirmationModalVariant =
+  | 'greyscale'
+  | 'blue'
+  | 'green'
+  | 'yellow'
+  | 'red';
+
+export interface ConfirmationModalProps extends Omit<DialogProps, 'title'> {
+  /* sets color of buttons */
+  variant?: ConfirmationModalVariant;
+  /* sets whether modal is open */
+  open: boolean;
+  /* function called when modal is closed */
+  onClose: () => void;
+  /* optional function called when modal is opened */
```

Comment:
nit: add an `@optional` here plz ([ex in Banner](https://github.com/shogun-hearth/hajimari/blob/0f37b6332a7868ab8514c61fba03d28c6b2de666/src/components/Banner/Banner.tsx#L32))